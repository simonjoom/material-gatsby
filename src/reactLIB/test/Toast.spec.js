import React from 'react';
import { shallow } from 'enzyme';
import Toast from '../src/Toast';
import mocker from './helper/new-mocker';

describe('Toast', () => {
  const options = {
    html: 'foo',
    classes: 'bar'
  };

  const toastMock = jest.fn();
  const restore = mocker('toast', toastMock);

  afterAll(() => {
    restore();
  });

  const child = 'toast content';

  const wrapper = shallow(<Toast options={options}>{child}</Toast>);

  test('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should create a child element', () => {
    expect(wrapper.contains(child)).toEqual(true);
  });

  test('calls Toast with clicked with options', () => {
    wrapper.simulate('click');
    expect(toastMock).toHaveBeenCalledWith(options);
  });
});
