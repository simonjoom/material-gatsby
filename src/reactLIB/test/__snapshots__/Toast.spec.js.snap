// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders 1`] = `
<Button
  className="toast"
  node="button"
  onClick={[Function]}
>
  toast content
</Button>
`;
